libpromises/files_names.c

```diff
@@ -170,6 +170,40 @@ char *JoinSuffix(char *path, size_t path_size, const char *leaf)
     return path;
 }
 
+char *JoinPaths(char *path, size_t path_size, const char *leaf_path)
+/*
+Just like the JoinSuffix() above, but makes sure there's a FILE_SEPARATOR
+between @path and @leaf_path.
+*/
+{
+    size_t len = strlen(leaf_path);
+    size_t path_len = strlen(path);
+
+    if (Chop(path, path_size) == -1)
+    {
+        Log(LOG_LEVEL_ERR, "Chop was called on a string that seemed to have no terminator");
+    }
+
+    if (path_len + len + 1 > path_size)
+    {
+        Log(LOG_LEVEL_ERR, "JoinPaths: Internal limit reached. Tried to add %s to %s",
+              leaf_path, path);
+        return NULL;
+    }
+
+    /* make sure there's a FILE_SEPARATOR between path and leaf_path */
+    if (!IsFileSep(path[path_len - 1]) && !IsFileSep(leaf_path[0]))
```

Comment:
what if `path_len` is zero

Reply:
Good catch! I'll also add tests for this case.